crates/language_server/src/convert.rs

```diff
@@ -102,19 +103,18 @@ pub(crate) mod diag {
     }
 
     impl IntoLspDiagnostic<'_> for &LoadingProblem<'_> {
-        type Feed = ();
-
-        fn into_lsp_diagnostic(self, _feed: &()) -> Option<Diagnostic> {
-            let range = Range {
-                start: Position {
-                    line: 0,
-                    character: 0,
-                },
-                end: Position {
-                    line: 0,
-                    character: 1,
-                },
-            };
+        type Feed = LineInfo;
+
+        fn into_lsp_diagnostic(self, line_info: &LineInfo) -> Option<Diagnostic> {
+            let range = self
+                .get_region()
+                .unwrap_or(Region::new(
+                    roc_region::all::Position::new(0),
+                    roc_region::all::Position::new(10000000),
```

Comment:
This will presumably make the whole screen red if we don't have a region. Maybe just the first char is a better default? If not, then maybe we should use `usize::MAX` instead of "some really big number".

Reply:
USize max is a bad plan. Lsp and json goes to javascript which, we must remember stores numbers as 64bit float and that would overflow on a usezie from 64bit systems. 

This is stupid large and should be fine in all practical applications, but I guess I could use f64max.

We do this because the main cases where we don't have a target it's because the whole document is cooked in some way. 

As Anton mentioned in zulip, it's often hard to find these errors if they are just on line 1 or something. I agree. If the error is a blocker for even parsing which precludes any other errors, why not show the whole doc as an erorr

Reply:
But I guess we don't care right, because it won't actually overflow it'll just be imprecise.. I still don't entirely trust it to handle super large numbers maybe I'm just too conservative 😅

I guess I worry the imprecision will do funky things.

Reply:
My arbitrary number should be about 10million chars is about 100k lines of code which is as much as anyone should ever need... 10k might be a little too small.